app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Api::V1::CommentsController < ApplicationController
+  before_action :set_post, only: %i[index create]
+  skip_before_action :verify_authenticity_token
+
+  # GET /api/v1/users/:user_id/posts/:post_id/comments
+  def index
+    @comments = @post.comments
+    render json: @comments
+  end
+
+  # POST /api/v1/posts/:post_id/comments
+
+  def create
+    if current_user.nil?
+      render json: { errors: ['Authentication required'] }, status: :unauthorized
+      return
+    end
+    @comment = @post.comments.create(comment_params.merge(user: current_user))
+
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity
+
+    end
+  end
```

Comment:
- You have done well so far, but please note that the authentication required for your rails API is different to the normal authentication, therefore the `current_user` method would be invalid. Please setup a proper authentication for your rails API, I suggest using [devise-token](https://github.com/baschtl/devise-token_authenticatable) or [devise-jwt](https://github.com/waiting-for-dev/devise-jwt) for this 👍 👍